zkevm-circuits/src/tx_circuit/sign_verify.rs

```diff
@@ -400,141 +348,213 @@ impl<F: Field, const MAX_VERIF: usize> SignVerifyChip<F, MAX_VERIF> {
         })
     }
 
+    #[allow(clippy::too_many_arguments)]
+    fn assign_rlc_le(
+        &self,
+        config: &SignVerifyConfig,
+        ctx: &mut RegionCtx<F>,
+        chips: &ChipsRef<F, NUMBER_OF_LIMBS, BIT_LEN_LIMB>,
+        name: &str,
+        q_rlc: Selector,
+        challenge: Value<F>,
+        inputs_le: impl IntoIterator<Item = Term<F>>,
+    ) -> Result<AssignedCell<F, F>, Error> {
+        let zero = chips.main_gate.assign_constant(ctx, F::zero())?;
+        let columns = config.main_gate_config.advices();
+        let inputs_le = inputs_le.into_iter().collect_vec();
+        let inputs_be = iter::repeat_with(|| Term::assigned(zero.cell(), Value::known(F::zero())))
+            .take(Integer::next_multiple_of(&inputs_le.len(), &columns.len()) - inputs_le.len())
+            .chain(inputs_le.into_iter().rev())
+            .collect_vec();
+
+        let mut rlc = challenge.map(|_| F::zero());
+        for (chunk_idx, chunk) in inputs_be.chunks_exact(columns.len()).enumerate() {
+            ctx.enable(q_rlc)?;
+            let assigned_rlc = ctx.assign_advice(|| "{name}_rlc[{chunk_idx}]", config.rlc, rlc)?;
+            for ((idx, column), term) in (chunk_idx * chunk.len()..).zip(columns).zip(chunk) {
+                let copied =
+                    ctx.assign_advice(|| format!("{name}_byte[{idx}]"), column, term.value())?;
+                if let Some(cell) = term.cell() {
+                    ctx.constrain_equal(cell, copied.cell())?;
```

Comment:
Let me know if I understand this correctly:
Whenever the `inputs_le` is not multiple of 5 (the number of columns in a main gate row), there are some front-padding 0s.  With this you are doing a copy constraint of those main gate cells, to a constant-zero cell right?  And afterwards the copy constraint is applied to the bytes.

Reply:
Exactly, since the `width` of `MainGate` not always divides length of `inputs_le`, so we need to add zero padding in front of `input_be` by copy constraint.